File: lc-files/_js/lc709.js
const toLowerCaseWithASCII = (str) => {
    let pointer = 0;
    let newStr = [];
    while(pointer < str.length) {
        const currCharCode = str.charCodeAt(pointer);
        if(currCharCode >= 65 && currCharCode <= 90) {
            newStr.push(String.fromCharCode(currCharCode + 32));
        } else {
            newStr.push(str.charAt(pointer));
        }
        pointer++;
    }

    return newStr.join("");
}

File: lc-files/python/lc709.py
def toLowerCaseWithASCII(str):
    pointer = 0;
    newStrArr = [];

    while(pointer < len(str)):
        currCharCode = ord(str[pointer]);
        if(currCharCode >= 65 and currCharCode <= 90):
            newStrArr.insert(chr(currCharCode + 32));
        else:
            newStrArr.insert(str[pointer]);

    return newStrArr.join("");
